main-app/app/scripts/player-select/player-service/player-service.js

```diff
@@ -0,0 +1,27 @@
+angular.module('Server.MyModule')
+    .service('PlayerService',  function () {
+        var me= this;
+        me.playerArray = ["human", "random", "pre-trained"];
+        me.player1 = me.playerArray[0];
+        me.player2 = me.playerArray[0];
+        var selectCharacter = function (playerChoice) {
+            if (playerChoice === me.playerArray[0]) {
+                return me.playerArray[1];
+            }
+            else if (playerChoice === me.playerArray[1]) {
+                return me.playerArray[2];
+            }
+            else {
+                return me.playerArray[0];
+            }
+        };
+
+        me.heroSelection = function (playerChoice) {
```

Comment:
I'd separate this down into two functions - togglePlayer1 and togglePlayer2 - removes the condition, while making the binding to this function clearer.